addons/mail/static/src/models/emoji_grid_view.js

```diff
@@ -16,12 +16,28 @@ registerModel({
             if (!this.emojiPickerViewOwner.emojiCategoryBarView.activeCategoryView) {
                 return clear();
             }
+            if (this.emojiPickerViewOwner.emojiSearchBar.currentSearch !== "") {
+                const filtered_emojis = this.emojiPickerViewOwner.emojiCategoryBarView.activeCategoryView.emojiCategory.allEmojis.filter(this._filterEmoji);
+                return insertAndReplace(
+                    filtered_emojis.map(emoji => {
+                        return { emoji: replace(emoji) };
+                    })
+                );
```

Comment:
Note that this search is quite inefficient.
If there's no perceptible performance issues, then keep it that way.

A more efficient way would require far more modeling than we have now, including knowing exactly which diffing change happen in the text input, so that we know exactly which emojis are affected or not, and so recompute only changing emojis.

But all this is too much work, so if no perceptible performance issues, then ignore.